### PR TITLE
cfg: replace std feature gate alloc imports with extern crate alloc

### PR DESCRIPTION
**Pull request by tcoratger**
Related https://github.com/paradigmxyz/reth/issues/10853

### CODE REVIEW

File: crates/primitives-traits/src/error.rs [hunk 1]
```diff
@@ -1,13 +1,11 @@
+extern crate alloc;
```
Comment (mattsse) [inline crates/primitives-traits/src/error.rs:1]: this should actually be in lib.rs

File: crates/revm/src/batch.rs [hunk 2]
```diff
@@ -182,15 +180,12 @@ impl BlockBatchRecord {
 
 #[cfg(test)]
 mod tests {
+    extern crate alloc;
```
Comment (mattsse) [inline crates/revm/src/batch.rs:183]: why is this required here? if we use alloc directly above

**Review by mattsse — CHANGES_REQUESTED** [review]
smol nit and 1q

File: crates/primitives-traits/src/error.rs (hunk 1 above)
Comment (tcoratger) [reply crates/primitives-traits/src/error.rs]: fixed

File: crates/revm/src/batch.rs (hunk 2 above)
Comment (tcoratger) [reply crates/revm/src/batch.rs]: indeed this is not needed, my bad, just fixed

File: crates/primitives/src/compression/mod.rs [hunk 3]
```diff
@@ -1,9 +1,7 @@
+use alloc::vec::Vec;
 use std::{cell::RefCell, thread_local};
```
Comment (emhane) [inline crates/primitives/src/compression/mod.rs:2]: what about this?

**Review by emhane — APPROVED** [review]
lgtm

File: crates/primitives/src/compression/mod.rs (hunk 3 above)
Comment (tcoratger) [reply crates/primitives/src/compression/mod.rs:2]: yeah I found this one, seems strange that is passes the no-std compilation before no?

Comment (emhane) [reply crates/primitives/src/compression/mod.rs:2]: yepp for sure weird

Comment (tcoratger) [reply crates/primitives/src/compression/mod.rs:2]: For Refcell I think we should use `use core::cell::RefCell;` but for `thread_local`, I don't think this is available so maybe we should keep as is no?

Comment (emhane) [reply crates/primitives/src/compression/mod.rs:2]: yepp, would be great if you can open an issue for replacing `thread_local` with smthg that works

Comment (tcoratger) [reply crates/primitives/src/compression/mod.rs]: - Fixed RefCell
- Created the issue https://github.com/paradigmxyz/reth/issues/10871

Comment (emhane) [reply crates/primitives/src/compression/mod.rs]: thanks

**Review by mattsse — APPROVED** [review]
